### PR TITLE
ACL support for health checks controller

### PR DESCRIPTION
**Pull request by kschoche**
This PR adds support for using ACLs with the health check controller.

4 minor changes are required:
1. `server-acl-init-job` must now generate a different set of rules depending on whether consulNamespaces/ACLs+HealthChecks/neither are defined. This is proposed as a non-required flag to `server-acl-init` to generate connect-inject's ACL tokens called `-enable-health-checks-controller`.
The new rules are defined in the incoming consul-k8s PR, but are 
```
  node_prefix "" {
     policy = "write"
  }
  service_prefix "" {
     policy = "write"
  }
```

2. update the `connect-inject-clusterrole` so that it can access secrets if ACLs+healthchecks are enabled.
3. new bats tests to cover the new flag and templating options.
4. the existing acceptance tests now pass ACL enabled/disabled and requisite intentions are added for those cases similarly to the existing connect_inject_test.

Note to reviewers : I fully expect some discussion on the naming of the flag and am open to ideas.

### CODE REVIEW

File: templates/connect-inject-deployment.yaml [hunk 1]
```diff
@@ -252,6 +252,7 @@ spec:
             consul-k8s acl-init \
               -secret-name="{{ template "consul.fullname" . }}-connect-inject-acl-token" \
               -k8s-namespace={{ .Release.Namespace }}
+
```
Comment (kschoche) [inline templates/connect-inject-deployment.yaml:255]:
```suggestion
```

File: test/acceptance/tests/connect/health_checks_test.go [hunk 2]
```diff
@@ -40,10 +43,12 @@ func TestHealthChecks(t *testing.T) {
 			cfg := suite.Config()
 
 			helmValues := map[string]string{
+				"global.imageK8S":                    "kschoche/consul-k8s-dev",
```
Comment (kschoche) [inline test/acceptance/tests/connect/health_checks_test.go:46]: this will be changed once the consul-k8s bits are in

**Comment by ashwin-venkatesh** [comment]
Is there a companion `consul-k8s` PR for this one that I could look at?

**Comment by kschoche** [comment]
@ashwin-venkatesh  yup! : https://github.com/hashicorp/consul-k8s/pull/368

File: templates/server-acl-init-job.yaml [hunk 3]
```diff
@@ -101,6 +101,10 @@ spec:
                 -resource-prefix=${CONSUL_FULLNAME} \
                 -k8s-namespace={{ .Release.Namespace }} \
 
+                {{- if (and .Values.connectInject.enabled .Values.connectInject.healthChecks.enabled) }}
+                -enable-health-checks-controller=true \
```
Comment (ashwin-venkatesh) [inline templates/server-acl-init-job.yaml:105]: Out of curiosity, do we have to set this value to be `true`? Can we just pass `-enable-health-checks-controller` and the presence of the flag imply truthiness?

File: test/acceptance/tests/connect/health_checks_test.go [hunk 4]
```diff
@@ -5,6 +5,9 @@ import (
 	"strconv"
 	"testing"
 
+	"github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/require"
+
```
Comment (ashwin-venkatesh) [inline test/acceptance/tests/connect/health_checks_test.go:10]:
```suggestion
```

**Review by thisisnotashwin — COMMENTED** [review]
This looks great. Just had a question about how we set the flag

File: templates/server-acl-init-job.yaml (hunk 3 above)
Comment (kschoche) [reply templates/server-acl-init-job.yaml:105]: pretty sure this is doable! when I refactor with the new naming I'll make this the case :)

**Review by thisisnotashwin — APPROVED** [review]
💯

**Review by lkysow — APPROVED** [review]
Once https://github.com/hashicorp/consul-k8s/pull/368 is merged.

File: .circleci/config.yml [hunk 5]
```diff
@@ -213,7 +213,7 @@ jobs:
               -kubeconfig="$HOME/.kube/$OC_PRIMARY_NAME" \
               -secondary-kubeconfig="$HOME/.kube/$OC_SECONDARY_NAME" \
               -debug-directory="$TEST_RESULTS/debug" \
-              -consul-k8s-image=hashicorpdev/consul-k8s:latest
+              -consul-k8s-image=kschoche/consul-k8s-dev
```
Comment (kschoche) [inline .circleci/config.yml:216]:
```suggestion
              -consul-k8s-image=hashicorpdev/consul-k8s:latest
```

File: test/acceptance/tests/connect/connect_inject_test.go [hunk 6]
```diff
@@ -32,6 +32,7 @@ func TestConnectInject(t *testing.T) {
 			ctx := suite.Environment().DefaultContext(t)
 
 			helmValues := map[string]string{
+				"global.imageK8S":              "kschoche/consul-k8s-dev",
```
Comment (kschoche) [inline test/acceptance/tests/connect/connect_inject_test.go:35]:
```suggestion
```

File: test/acceptance/tests/connect/health_checks_test.go [hunk 7]
```diff
@@ -40,10 +42,12 @@ func TestHealthChecks(t *testing.T) {
 			cfg := suite.Config()
 
 			helmValues := map[string]string{
+				"global.imageK8S":                    "kschoche/consul-k8s-dev",
```
Comment (kschoche) [inline test/acceptance/tests/connect/health_checks_test.go:45]:
```suggestion
```

**Pull request merged**